agent/api/containerstatus.go

```diff
@@ -22,6 +22,13 @@ const (
 	ContainerCreated
 	// ContainerRunning represents a container that has started
 	ContainerRunning
+	// ContainerResourcesProvisioned represents a container that has completed provisioning all of its
+	// resources. Non-internal containers (containers present in the task definition) transition to
```

Comment:
That is a valid concern. The way the task engine is structured, it's *very tricky* to get per container transition rules, where we can specify one set of containers to go through a given set of transitions and the other set go through this other set of transitions. I'm not sure how tricky it'd be to set the steady state status to `RUNNING` for non ENI tasks and to `RESOURCES_READY` for ENI tasks. Let me know if that sounds like a better alternative.